### DISCUSSION TITLE
How to copy files (*.resx, *.ico, ...) from buildPath to buildDirectory

### DISCUSSION THREAD

**Discussion by ccecvb** [vscode-abl/vscode-abl#172] [Q&A]
Certain files in the buildpath are required in the build target to run the application without sources in the propath.

These files include
*.ico *.resx *.wrx

In our build script I use the target list below.

How can I achieve this in vscode ?

Is there an existing feature that covers this ?

Could this be added to the buildpath ?

### ant target

``` xml
    <target name="copy_resources"
        description="Copy bmp gif ico jpg json png resx and wrx from projects to rcodedir"
        extensionOf="build">
        <echo message="Copy from ${projects_src_list} to ${rcodedir}" />
        <copy todir="${rcodedir}">

            <multirootfileset basedirs="${projects_src_list}" casesensitive="false"
                erroronmissingdir="false">
                <include name="**/*.annotations" />
                <include name="**/*.bat" />
                <include name="**/*.bmp" />
                <include name="**/*.css" />
                <include name="**/*.csv" />
                <include name="**/*.dll" />
                <include name="**/*.gif" />
                <include name="**/*.html" />
                <include name="**/*.ico" />
                <include name="**/*.jpg" />
                <include name="**/*.JPG" />
                <include name="**/*.json" />
                <include name="**/*.png" />
                <include name="**/*.resx" />
                <include name="**/*.sh" />
                <include name="**/*.txt" />
                <include name="**/*.wrx" />
                <include name="**/*.wsdl" />
                <include name="**/*.wsm" />
                <include name="**/*.xml" />
                <include name="**/*.xsd" />
            </multirootfileset>
        </copy>
    </target>

```

**Comment by stefandrissen** [comment]
As an indirect answer, there is a VS code plug in that allows execution of Ant targets - https://github.com/NickHeap2/vscode-ant

**Reply by ccecvb** [reply]
Thanks Stefan, I'll look into it

**Comment by gquerret** [comment]
The language server does not copy resources, and I'm not sure there's any value doing plain copy of files from `src/` to `build/`.
What if you had the `src` directory in your execution propath after `build/` ?

**Reply by cverbiest** [reply]
In pdsoe we compile to src, there we have all src propath entries in the runtime propath.

 In vscode we currently compile all src directories into a single tty or gui target, a runtime client would only need that directory in it's propath if that also contains resx etc.

We would like to be able to start a runtime session that resembles the real runtime from our build process.

I'll have a look at the ant builder

**Reply by gquerret** [reply]
And wouldn't it be easier to have a `resources` directory that would be entirely included in the package ? That would also avoid having to specify the patterns to be included, you just add everything.
